src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -26,6 +26,12 @@ def get_location_completion_list(prefix, **kwargs):#pylint: disable=unused-argum
     result = get_subscription_locations()
     return [l.name for l in result]
 
+def location_name_type(name):
+    if ' ' in name:
+        # if display name is provided, attempt to convert to short form name
+        name = next((l.name for l in get_subscription_locations() if l.display_name == name), name)
```

Comment:
@tjprescott, also ensure to update similar logic in [validators](https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L64). I think we are good now by accepting both naming styles. We can call it done after that. I see no need to port xplat issue, unless you have other work to do.